### PR TITLE
CI Improvements

### PR DESCRIPTION
**Pull request by stewartie4**
- remove hardcoded branch variable
- remove self hosted label from system tests

### CODE REVIEW

File: deploy-0chain/action.yml [hunk 1]
```diff
@@ -69,21 +69,13 @@ inputs:
...
-#          ================================
-
 runs:
   using: "composite"
   steps:
     - name: "Setup Config"
       shell: 'script --return --quiet --command "bash {0}"'
       run: |
-        echo "ACTIONS_BRANCH=master" >> $GITHUB_ENV
+        echo "ACTIONS_BRANCH=$(cd ../../_actions/0chain/actions/ && find . -type d  -maxdepth 2 | awk -F/ '{print NF "\t" $0}' | sort -t: -k1 -g -r | head -1 | cut -f2 | cut -c 3-)" >> $GITHUB_ENV
```
Comment (stewartie4) [inline deploy-0chain/action.yml:78]: Yes, I know. lol
This monster of a command derives the current actions branch name by listing the directory used to checkout the action

**Comment by stewartie4** [comment]
Don't merge for now - i found a bug :(

File: deploy-0chain/action.yml (hunk 1 above)
Comment (stewartie4) [reply deploy-0chain/action.yml]: replaced the above command with something less brittle, should be good to merge now

**Review by Kishan-Dhakan — APPROVED** [review]
Great to get this in!

**Pull request merged**